tests/drum/test_fit.py

```diff
@@ -369,6 +367,33 @@ def test_transform_fit_fails_default(
             )
             assert "WARNING: No type schema provided. For transforms, we" not in stdout
 
+    @pytest.mark.parametrize(
+        "framework, language", [(SKLEARN_TRANSFORM_WITH_Y, PYTHON), (R_TRANSFORM_NO_Y, R_FIT),]
+    )
+    @pytest.mark.parametrize("problem", [REGRESSION, BINARY, ANOMALY])
+    def test_transform_fit_disallow_y_output(
+        self, resources, tmp_path, framework, language, problem
+    ):
+
+        input_dataset = resources.datasets(SKLEARN, TRANSFORM)
+        target_type = TRANSFORM
+        custom_model_dir = _create_custom_model_dir(
+            resources, tmp_path, framework, problem, language=framework,
+        )
+
+        cmd = "{} fit --target-type {} --code-dir {} --input {} --verbose {}".format(
+            ArgumentsOptions.MAIN_COMMAND,
+            target_type,
+            custom_model_dir,
+            input_dataset,
+            "--disable-strict-validation",
+        )
+        with pytest.raises(AssertionError):
+            _, stdout, _ = _exec_shell_cmd(
+                cmd, "Failed in {} command line! {}".format(ArgumentsOptions.MAIN_COMMAND, cmd)
+            )
+            assert "Transformation of the target variable is not supported by DRUM." in stdout
```

Comment:
This assertion will not be reached. You need to add `assert_if_fail=False` to the _exec call above